### PR TITLE
Naming conventions

### PR DESCRIPTION
**Pull request by abandeali1**
## Description

- This mostly prefixes `internal` and `private` functions with an `_` (except for libraries)
- A few other variables are renamed for consistency or clarity
- Some unused files/functions are removed

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/asset-proxy/contracts/src/ERC1155Proxy.sol [hunk 1]
```diff
@@ -28,6 +28,7 @@ contract ERC1155Proxy is
     // Id of this proxy.
     bytes4 constant internal PROXY_ID = bytes4(keccak256("ERC1155Assets(address,uint256[],uint256[],bytes)"));
 
+    // solhint-disable-next-line payable-fallback
```
Comment (feuGeneA) [inline contracts/asset-proxy/contracts/src/ERC1155Proxy.sol]: :pray:

File: contracts/utils/contracts/src/SafeMath.sol [hunk 2]
```diff
@@ -53,31 +53,15 @@ contract SafeMath {
         return c;
     }
 
-    function max64(uint64 a, uint64 b)
```
Comment (feuGeneA) [inline contracts/utils/contracts/src/SafeMath.sol]: not handy?

**Comment by abandeali1** [comment]
I looked into a linting rule for this and couldn't find any for solhint unfortunately.

File: contracts/utils/contracts/src/SafeMath.sol (hunk 2 above)
Comment (abandeali1) [reply contracts/utils/contracts/src/SafeMath.sol]: We don't actually use `uint64` anywhere in our codebase AFAIK, not sure why this was there in the first place.

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 3]
```diff
@@ -28,112 +28,112 @@ contract LibExchangeRichErrorDecoder is
     /// @dev Decompose an ABI-encoded SignatureError.
     /// @param encoded ABI-encoded revert error.
     /// @return errorCode The error code.
-    /// @return signer The expected signer of the hash.
+    /// @return signerAddress The expected signerAddress of the hash.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return signerAddress The expected signer of the hash.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 4]
```diff
@@ -28,112 +28,112 @@ contract LibExchangeRichErrorDecoder is
...
         hash = _readErrorParameterAsBytes32(encoded, 1);
-        signer = _readErrorParameterAsAddress(encoded, 2);
+        signerAddress = _readErrorParameterAsAddress(encoded, 2);
         signature = _readErrorParameterAsBytes(encoded, 3);
     }
 
     /// @dev Decompose an ABI-encoded SignatureValidatorError.
     /// @param encoded ABI-encoded revert error.
-    /// @return signer The expected signer of the hash.
+    /// @return signerAddress The expected signerAddress of the hash.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return signerAddress The expected signer of the hash.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 5]
```diff
@@ -28,112 +28,112 @@ contract LibExchangeRichErrorDecoder is
...
+        signerAddress = _readErrorParameterAsAddress(encoded, 1);
         signature = _readErrorParameterAsBytes(encoded, 2);
         errorData = _readErrorParameterAsBytes(encoded, 3);
     }
 
     /// @dev Decompose an ABI-encoded SignatureWalletError.
     /// @param encoded ABI-encoded revert error.
     /// @return errorCode The error code.
-    /// @return signer The expected signer of the hash.
+    /// @return signerAddress The expected signerAddress of the hash.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return signerAddress The expected signer of the hash.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 6]
```diff
@@ -28,112 +28,112 @@ contract LibExchangeRichErrorDecoder is
...
+        signerAddress = _readErrorParameterAsAddress(encoded, 1);
         signature = _readErrorParameterAsBytes(encoded, 2);
         errorData = _readErrorParameterAsBytes(encoded, 3);
     }
 
     /// @dev Decompose an ABI-encoded SignatureOrderValidatorError.
     /// @param encoded ABI-encoded revert error.
     /// @return errorCode The error code.
-    /// @return signer The expected signer of the hash.
+    /// @return signerAddress The expected signerAddress of the hash.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return signerAddress The expected signer of the hash.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 7]
```diff
@@ -28,112 +28,112 @@ contract LibExchangeRichErrorDecoder is
...
+        signerAddress = _readErrorParameterAsAddress(encoded, 1);
         signature = _readErrorParameterAsBytes(encoded, 2);
         errorData = _readErrorParameterAsBytes(encoded, 3);
     }
 
     /// @dev Decompose an ABI-encoded SignatureWalletOrderValidatorError.
     /// @param encoded ABI-encoded revert error.
     /// @return errorCode The error code.
-    /// @return signer The expected signer of the hash.
+    /// @return signerAddress The expected signerAddress of the hash.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return signerAddress The expected signer of the hash.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 8]
```diff
@@ -158,52 +158,52 @@ contract LibExchangeRichErrorDecoder is
     /// @dev Decompose an ABI-encoded InvalidSenderError.
     /// @param encoded ABI-encoded revert error.
     /// @return orderHash The order hash.
-    /// @return sender The sender.
+    /// @return senderAddress The senderAddress.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return senderAddress The sender.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 9]
```diff
@@ -158,52 +158,52 @@ contract LibExchangeRichErrorDecoder is
...
         orderHash = _readErrorParameterAsBytes32(encoded, 0);
-        sender = _readErrorParameterAsAddress(encoded, 1);
+        senderAddress = _readErrorParameterAsAddress(encoded, 1);
     }
 
     /// @dev Decompose an ABI-encoded InvalidMakerError.
     /// @param encoded ABI-encoded revert error.
     /// @return orderHash The order hash.
-    /// @return maker The maker of the order.
+    /// @return makerAddress The makerAddress of the order.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return makerAddress The maker of the order.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 10]
```diff
@@ -158,52 +158,52 @@ contract LibExchangeRichErrorDecoder is
...
         orderHash = _readErrorParameterAsBytes32(encoded, 0);
-        maker = _readErrorParameterAsAddress(encoded, 1);
+        makerAddress = _readErrorParameterAsAddress(encoded, 1);
     }
 
     /// @dev Decompose an ABI-encoded InvalidTaker.
     /// @param encoded ABI-encoded revert error.
     /// @return orderHash The order hash.
-    /// @return taker The taker of the order.
+    /// @return takerAddress The takerAddress of the order.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return takerAddress The taker of the order.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 11]
```diff
@@ -225,21 +225,21 @@ contract LibExchangeRichErrorDecoder is
 
     /// @dev Decompose an ABI-encoded OrderEpochError.
     /// @param encoded ABI-encoded revert error.
-    /// @return maker The order maker.
-    /// @return sender The sender.
-    /// @return currentEpoch The current epoch for the maker.
+    /// @return makerAddress The order makerAddress.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return makerAddress The order maker.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 12]
```diff
@@ -225,21 +225,21 @@ contract LibExchangeRichErrorDecoder is
...
+    /// @return orderSenderAddress The senderAddress.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return orderSenderAddress The order sender.
```

File: contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol [hunk 13]
```diff
@@ -225,21 +225,21 @@ contract LibExchangeRichErrorDecoder is
...
+    /// @return currentEpoch The current epoch for the makerAddress.
```
Comment (dorothy-zbornak) [inline contracts/exchange/contracts/src/libs/LibExchangeRichErrorDecoder.sol]:
```suggestion
    /// @return currentEpoch The current epoch for the maker.
```

File: contracts/multisig/contracts/src/MultiSigWallet.sol [hunk 14]
```diff
@@ -242,7 +242,7 @@ contract MultiSigWallet {
 
     // call has been separated into its own function in order to take advantage
     // of the Solidity's code generator to produce a loop that copies tx.data into memory.
-    function external_call(address destination, uint value, uint dataLength, bytes data) internal returns (bool) {
+    function _external_call(address destination, uint value, uint dataLength, bytes data) internal returns (bool) {
```
Comment (hysz) [inline contracts/multisig/contracts/src/MultiSigWallet.sol]: Should we make this `_externalCall` ?

**Comment by abandeali1** [comment]
@dorothy-zbornak the Address library change was a mistake and has been fixed. I also renamed `Address` to `LibAddress` for consistency.

**Pull request merged**